### PR TITLE
dnos9_template (New Module)

### PR DESCRIPTION
**Pull request by skg-net**
##### ISSUE TYPE
- New Module Pull Request
##### COMPONENT NAME
- modules/core/network/dnos9/dnos9_template
##### ANSIBLE VERSION
- ansible 2.2.0 (devel a194740) last updated 2016/09/08 17:44:28 (GMT -700) lib/ansible/modules/core: (detached HEAD db38f0c876) last updated 2016/09/08 17:44:58 (GMT -700) lib/ansible/modules/extras: (detached HEAD 8bfdcfcab2) last updated 2016/09/08 17:44:58 (GMT -700) config file = /home/skg/lothlorien/ansible.cfg configured module search path = Default w/o overrides
  ##### SUMMARY
- Added support for dnos9_template module to support Dell Networking OS9 devices.

### CODE REVIEW

**Comment by skg-net** [comment]
@privateip @gundalow, please have a look at it.

**Comment by gregdek** [comment]
Thanks @skg-net for this new module. When this module receives 'shipit' comments from two community members and any 'needs_revision' comments have been resolved, we will mark for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

File: network/dnos9/dnos9_template.py [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+        file in role or playbook root folder in templates unless a full
+        path to the file is given.
+    required: true
+  force:
+    description:
+      - The force argument instructs the module not to consider the
+        current device running-config.  When set to true, this will
+        cause the module to push the contents of I(src) into the device
+        without first checking if already configured. This argument is
+        mutually exclusive with O(config).
```
Comment (gundalow) [inline network/dnos9/dnos9_template.py]: `I(...)`

File: network/dnos9/dnos9_template.py [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+    required: false
+    default: false
+    choices: [ "true", "false" ]
+  backup:
+    description:
+      - When this argument is configured true, the module will backup
+        the running-config from the node prior to making any changes.
+        The backup file will be written to backup_{{ hostname }} in
+        the root of the playbook directory. This argument is
+        mutually exclusive with O(config).
```
Comment (gundalow) [inline network/dnos9/dnos9_template.py]: `I(...)`

File: network/dnos9/dnos9_template.py [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+  config:
+    description:
+      - The module, by default, will connect to the remote device and
+        retrieve the current running-config to use as a base for comparing
+        against the contents of source.  There are times when it is not
+        desirable to have the task get the current running-config for
+        every task.  The I(config) argument allows the implementer to
+        pass in the configuration to use as the base config for
+        comparison. This argument is mutually exclusive with
+        O(force) and O(backup).
```
Comment (gundalow) [inline network/dnos9/dnos9_template.py]: `I(...)` *2

**Comment by gundalow** [comment]
needs_revision

**Comment by gregdek** [comment]
Thanks @skg-net for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by skg-net** [comment]
ready_for_review

**Comment by gregdek** [comment]
Thanks @skg-net for this new module. When this module receives 'shipit' comments from two community members and any 'needs_revision' comments have been resolved, we will mark for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gundalow** [comment]
shipit

**Comment by gregdek** [comment]
Thanks again to @skg-net for this PR, and thanks @ansible/core for reviewing. Marking for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gundalow** [comment]
Merged, thanks

**Pull request merged**